src/main/kotlin/org/jetbrains/gradle/ext/CompilerExtension.kt

```diff
@@ -0,0 +1,34 @@
+package org.jetbrains.gradle.ext
+
+import org.gradle.api.plugins.ExtensionAware
+import org.gradle.kotlin.dsl.the
+import org.gradle.plugins.ide.idea.model.IdeaModel
+
+private fun IdeaModel.getCompiler(): IdeaCompilerConfiguration {
+    val settings = (this.project as ExtensionAware).the<ProjectSettings>() as ExtensionAware
+    return settings.the()
+}
+
+fun IdeaModel.withJavacAdditionalOptions(options: List<String>): IdeaModel {
+    this.getCompiler().apply {
+        javac {
+            javacAdditionalOptions = options.joinToString(separator = " ")
+        }
+    }
+    return this
+}
+
+fun IdeaModel.withParallelCompilation(parallelCompilation: Boolean): IdeaModel {
+    this.getCompiler().apply {
+        this.parallelCompilation = parallelCompilation
+    }
+    return this
+}
+
+fun IdeaModel.withProcessHeapSizeInMb(processHeapSizeInMb: Int): IdeaModel {
+    this.getCompiler().apply {
+        this.processHeapSize = processHeapSizeInMb
+    }
+    return this
+}
```

Comment:
I would do this a little bit differently so that it feels more natural to use. I would add an accessor for each object and a function accepting an action to configure it. You can see the concept in the test retry plugin: https://github.com/gradle/test-retry-gradle-plugin/blob/master/plugin/src/main/kotlin/org/gradle/kotlin/dsl/testRetry.kt

So for example:

```kotlin
val IdeaModel.settings: ProjectSettings 
    get() = (this.project as ExtensionAware).the()

fun IdeaModel.settings(configure: ProjectSettings.() -> Unit) {
    configure(this.settings)
}

val ProjectSettings.compiler: IdeaCompilerConfiguration
    get() = the()

fun ProjectSettings.compiler(configure: IdeaCompilerConfiguration.() -> Unit) {
   configure.apply(this.compiler)
}
```

Then in your build script you can write something like:

```kotlin
idea {
    settings {
        compiler {
           // configure the compiler
        }
    }
}
```

This is generated automatically by the Kotlin DSL plugin if you write the plugin in Kotlin. In this case it needs to be done by hand for all extensions you want to configure.

Reply:
Thanks for the suggestion, it looks more natural indeed 👌

Reply:
Small correction: accessors are only generated by the Kotlin DSL plugin for extensions on `Project`. So the Kotlin DSL plugin won't help you here and you need to do it manually.

Reply:
Not sure what you meant by that, but I pushed a new commit which allow to get configuration blocks rather than method calls:
```
configure<IdeaModel> {
    settings {
        compiler {
            processHeapSize = 4096
            parallelCompilation = false
            javac {
                javacAdditionalOptions = listOf(
                    "-Werror",
                    "-Xlint:all",
                    "-Xlint:-options",
                    "-Xlint:-serial",
                    "-proc:none",
                    "-Xdiags:verbose",
                    "-parameters"
                ).joinToString(separator = " ")
            }
        }
    }
}
```
Let me know what/if I'm missing something

Reply:
@jprinet looks awesome. Can't wait to use this. In your example can the `configure<IdeaModel> {` be replaced with `idea` if the plugin is applied alongside the core `idea` plugin?

Reply:
let me try 😅

Reply:
Guess what @britter, this works out of the box 😅 
The `idea` configuration block is taking an `IdeaModel` as parameter which is the one the _Kotlin_ extension where added for.